include/ismrmrd/serialize.h

```diff
@@ -190,12 +190,14 @@ void archive_data(Archive &ar, uint16_t data_type, void *data, size_t datasize)
 }
 
 template <class Archive>
-void save_helper(Archive &ar, ISMRMRD::ISMRMRD_ImageHeader const &hdr, void* data, size_t data_sz, char* attribute_string,
-                 size_t attr_str_sz, CompressionParameters const& params = CompressionParameters()) {
+void save_helper(Archive &ar, ISMRMRD::ISMRMRD_ImageHeader const &hdr, void* data, size_t data_sz, const char* attribute_string,
+    CompressionParameters const& params = CompressionParameters()) {
     ar(make_nvp("head", hdr));
-
-    if (hdr.attribute_string_len > 0)
-        ar(make_nvp("attribute_string", binary_data(attribute_string, attr_str_sz)));
+    if (hdr.attribute_string_len > 0) {
+        assert(hdr.attribute_string_len == strlen(attribute_string));
```

Comment:
This assert prevents you from having a null character in the attribute string, e.g. the character array is encrypted.